core/src/main/java/io/grpc/internal/MessageDeframer.java

```diff
@@ -258,9 +258,11 @@ private void deliver() {
       stalled = !isDataAvailable();
 
       if (endOfStream) {
-        if (!isDataAvailable()) {
+        if (!stalled) {
```

Comment:
The fix isn't correct either. You are inverting the meaning of stalled. The fix doesn't pass the tests just like the original commit.